angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -2,13 +2,40 @@ import { Component } from '@angular/core';
 import Storage from 'src/services/storage.service';
 import { IdentifierDataType } from 'src/utils/enumerations/identifier-data-type';
 
+enum BMS_FAULTS_TYPES {
+  CELLS_NOT_BALANCING = 1,
+  CELL_VOLTAGE_TOO_LOW = 2,
+  CELL_VOLTAGE_TOO_HIGH = 4,
+  PACK_TOO_HOT = 8,
+  OPEN_WIRING_FAULT = 16,
+  INTERNAL_SOFTWARE_FAULT = 32,
+  INTERNAL_THERMAL_ERROR = 64,
+  INTERNAL_CELL_COMM_FAULT = 128,
+  CURRENT_SENSOR_FAULT = 256,
+  CHARGE_READING_MISMATCH = 512,
+  LOW_CELL_VOLTAGE = 1024,
+  WEAK_PACK_FAULT = 2048,
+  EXTERNAL_CAN_FAULT = 4096,
+  DISCHARGE_LIMIT_ENFORCEMENT_FAULT = 8192,
+  CHARGER_SAFETY_RELAY = 16384,
+  BATTERY_THERMISTOR = 32768,
+  CHARGER_CAN_FAULT = 65536,
+  CHARGER_LIMIT_ENFORCEMENT_FAULT = 131072
+}
+
+enum FaultType {
+  BMS = 'BMS',
+  Charger = 'Charger'
+}
+
 @Component({
   selector: 'fault-display',
   templateUrl: './fault-display.component.html',
   styleUrls: ['./fault-display.component.css']
 })
 export default class FaultDisplay {
-  faults: { faultName: string; time: string }[] = [];
+  faults: { type: string; name: string; time: string }[] = [];
+  faultsShifted: boolean = false;
```

Comment:
You shouldn't need this variable

Reply:
That is to maintain consistent coloring of the faults(e.i. a orange fault will stay the same color as long as it is the top 50 faults(displayed)). if you think should that's unnecessary all for removing it.